### ISSUE TITLE
Wrong paths added to the project search path?

### ISSUE THREAD

**Issue by Spelt** [HashLoad/boss#156]
Hi

I manage ZXing for Delphi and I need a little help in order to support Boss properly. I have uploaded the boss.json here: https://github.com/Spelt/ZXing.Delphi/blob/v_3.0/boss.json

The problem is the line  "mainsrc": "Lib/Classes/",

It seems right but all other paths next to Lib are also added to the path. For example the this path is added: modules\ZXing.Delphi\UnitTest 

I would expect only everything in: modules\ZXing.Delphi\Lib\Classes

Or am I doing something wrong here?

Thanks!
Edward

**Comment by viniciussanchez** [comment]
Normally he takes the first folder... it could be a bug in the boss. Until then I always only worked with "/src" for example.

**Comment by andriwsluna** [comment]
@Spelt , i installed your packge here and boss added only de lib/classes folder and your sub folders: modules\ZXing.Delphi\Lib\Classes\1D Barcodes;
..\modules\ZXing.Delphi\Lib\Classes\1D Barcodes;
modules\ZXing.Delphi\Lib\Classes\2D Barcodes\Decoder;
..\modules\ZXing.Delphi\Lib\Classes\2D Barcodes\Decoder;
modules\ZXing.Delphi\Lib\Classes\2D Barcodes\Detector;
..\modules\ZXing.Delphi\Lib\Classes\2D Barcodes\Detector;
modules\ZXing.Delphi\Lib\Classes\2D Barcodes;
..\modules\ZXing.Delphi\Lib\Classes\2D Barcodes;
modules\ZXing.Delphi\Lib\Classes\Common\Detector;
..\modules\ZXing.Delphi\Lib\Classes\Common\Detector;
modules\ZXing.Delphi\Lib\Classes\Common\ReedSolomon;
..\modules\ZXing.Delphi\Lib\Classes\Common\ReedSolomon;
modules\ZXing.Delphi\Lib\Classes\Common;
..\modules\ZXing.Delphi\Lib\Classes\Common;
modules\ZXing.Delphi\Lib\Classes\Filtering;
..\modules\ZXing.Delphi\Lib\Classes\Filtering;
modules\ZXing.Delphi\Lib\Classes;
..\modules\ZXing.Delphi\Lib\Classes

The "modules\ZXing.Delphi\UnitTest" are not added. Try to upgrade boss version.

**Comment by Spelt** [comment]
Hi,

Thanks that works but I now see another problem. The folders are added
twice ! One with the right path and one with the wrong path.

Like:

modules\ZXing.Delphi\Lib\Classes\2D Barcodes\Decoder;
..\modules\ZXing.Delphi\Lib\Classes\2D Barcodes\Decoder;

**Comment by andriwsluna** [comment]
This is cause by this [PR](https://github.com/HashLoad/boss/pull/147/files). 
I don't know exactly why this adjustment

**Comment by Spelt** [comment]
Yes I think so too. Op 1 jun 2024 om 15:51 heeft Andriws Luna ***@***.***> het volgende geschreven:﻿
This is cause by this PR.
I don't know exactly why this adjustment

**Comment by birbilis** [comment]
> the folders are added
> twice ! One with the right path and one with the wrong path.
> 
> Like:
> 
> modules\ZXing.Delphi\Lib\Classes\2D Barcodes\Decoder;
> ..\modules\ZXing.Delphi\Lib\Classes\2D Barcodes\Decoder;

I see the same issue, then one gets a ton of build warnings cause of that

if they really needed to prefix something they could have used .\

![Image](https://github.com/user-attachments/assets/041abedd-5bdd-4f27-819c-7e67085db7a6)

![Image](https://github.com/user-attachments/assets/5b2b560a-dfb8-48f0-b490-1998edb4506f)

![Image](https://github.com/user-attachments/assets/9c9e5535-2faa-40af-ac96-a273ac99179f)
